docs/reference-manual/native-image/NMT.md

```diff
@@ -0,0 +1,75 @@
+## Limitations
+
+In OpenJDK, NMT has two modes: summary and detailed. Currently in Native Image, only NMT summary mode is supported. Detailed mode, which enables callsite tracking, is not available. Capturing baselines are also not yet possible. If you are interested in support for these additional features, please make a request on the GraalVM Github project. 
+
+Only malloc tracking is available in GraalVM for JDK 23. Virtual memory tracking is available in early access releases. 
+
+A limitation Native Image NMT shares with OpenJDK is that it can only track allocations at the VM-level and those made with `Unsafe#allocateMemory(long)`. For example, if library code or application code calls malloc directly, that call will bypass the NMT accounting and be untracked.
+
+## Performance
+In most cases, both the CPU and memory overhead of NMT will be quite minimal. There is an overhead of 16B per malloc allocation (to accommodate malloc headers) that is reclaimed once the memory is freed. In practice this does not amount to very much since SubstrateVM does not make many allocations. One can usually expect less than a thousand allocations at any point in time. This overhead scales with allocation count, not size, so if `Unsafe#allocateMemory(long)` is frequently used at the application level, the overhead may become noticeable. However, this case is unlikely.
+
+There is also minimal CPU overhead since not much work needs to be done internally to track allocations. Synchronization is also minimal apart from atomic counters.
+
+In comparison to other serviceability features such as JFR, NMT has relatively very little overhead.
+
+## JFR Events for NMT
+The OpenJDK JFR events `jdk.NativeMemoryUsage` and `jdk.NativeMemoryUsageTotal` are supported in Native Image.
+
+There are also two Native Image specific JFR events that you can access: `jdk.NativeMemoryUsagePeak` and `jdk.NativeMemoryUsageTotalPeak`. These Native Image specific events have been created to expose peak usage data otherwise not exposed through the JFR events ported over from the OpenJDK. It should be noted that these new are marked as experimental. You may need to enable experimental events in software like JDK Mission control in order to view them.
+
+Below is example of what the new events look like viewed using the `jfr` command line tool:
+```
+$jfr print --events jdk.NativeMemoryUsagePeak recording.jfr 
+
+jdk.NativeMemoryUsagePeak {
+  startTime = 13:18:50.605 (2024-04-30)
+  type = "Threading"
+  peakReserved = 424 bytes
+  peakCommitted = 424 bytes
+  countAtPeak = 4
+  eventThread = "JFR Shutdown Hook" (javaThreadId = 63)
+}
+
+jdk.NativeMemoryUsagePeak {
+  startTime = 13:18:50.605 (2024-04-30)
+  type = "Unsafe"
+  peakReserved = 14.0 kB
+  peakCommitted = 14.0 kB
+  countAtPeak = 2
+  eventThread = "JFR Shutdown Hook" (javaThreadId = 63)
+}
+```
+To use these JFR events for NMT, JFR must be included as a feature at build time and started at runtime. 
+
+### Further Reading
+
```

Comment:
Add one more bullet point:
```markdown
 - [Debugging and Diagnostics](DebuggingAndDiagnostics.md)
```

Reply:
@roberttoyonaga, could you please also add a link to "Native Memory Tracking (NMT) with Native Image" in the DebuggingAndDiagnostics.md file?